database/tables/group_users.pg

```diff
@@ -1,10 +1,10 @@
 columns
     group_id: bigint not null
-    group_role_id: bigint
+    group_role_id: bigint not null
     user_id: bigint not null
 
 indexes
-    group_users_pkey: PRIMARY KEY (group_id, user_id) USING btree (group_id, user_id)
+    group_users_pkey: PRIMARY KEY (group_id, user_id, group_role_id) USING btree (group_id, user_id, group_role_id)
```

Comment:
Does including the role in the key imply that a given user may have multiple roles for a given group?

Reply:
Yes, a given user can have multiple roles in any given group. This allows groups to continue completing assessments with the necessary roles even if users leave the group while completing the assessment.

This feature is partially why we had the code-smell value that you pointed out [here](https://github.com/PrairieLearn/PrairieLearn/pull/6109#discussion_r1037628409). We weren't sure exactly how to implement multiple roles per user in a group while also ensuring that `group_users` could still represent users in groups that don't use group roles. We'll try and think of a workaround, but any advice would be greatly appreciated!

Reply:
It sounds like this has been solved by moving role assignments to another table! Feel free to resolve if that's the case.